modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx

```diff
@@ -77,13 +77,33 @@ function FormPagination (props) {
   let [ activePage, setActivePage ] = useState(0);
   let [ direction, setDirection ] = useState(1);
   let [ nextActivePage, setNextActivePage ] = useState();
+  let [ numberOfSteps, setNumberOfSteps ] = useState();
+  let [ activeStep, setActiveStep ] = useState();
+  let [ progress, setProgress ] = useState(2);
   const DIRECTION_NEXT = 1, DIRECTION_PREV = -1;
 
   let previousEntryType;
   let questionIndex = 0;
   let pagesResults = {};
   let pagesArray = [];
 
+  useEffect(() => {
+    let newNumberOfSteps = 0;
+    let newActiveStep = 0;
+    pages.forEach(page => {
+      if (page.canBeVisible) {
+        newNumberOfSteps++;
+      }
+    })
```

Comment:
Can we make it less verbose?

```suggestion
    pages.forEach(page => page.canBeVisible && newNumberOfSteps++)
```

Reply:
Or even
```suggestion
    setNumberOfSteps(pages.filter(page => page.canBeVisible).length);
```